teps/0008-support-knative-service-for-triggers-eventlistener-pod.md

```diff
@@ -0,0 +1,318 @@
+kubernetes deployment with default values and yaml looks something like below.
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+```
+
+#### Knative Service `OR any CRD`
+To support Knative Service along with Kubernetes Deployment we use dynamic `Raw` data so that it can be any CRD like `serving.knative.dev.` 
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+ service:
+   dynamic:
+     apiVersion: serving.knative.dev/v1  #It can be any CRD (foo.bar.com)
+     kind: Service
+     metadata:
+       labels:
+         serving.knative.dev/visibility: "cluster-local"
+     spec:
+       template:
+         metadata:
+           annotations:
+             autoscaling.knative.dev/minScale: "1"
+         spec:
+           serviceAccountName: tekton-triggers-github-sa
+           nodeSelector:
+             app: test
+           tolerations:
+           - key: key
+             value: value
+             operator: Equal
+             effect: NoSchedule
+```
+
+## Design Details
+
+The main goal of this TEP is to make triggers flexible enough to accept any CRD in order to create an eventlistener pod.
+
+Kubernetes Deployment, Knative Service both have [PodSpec](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3704) as a common sub-field so usage of
+[WithPod{}](https://github.com/knative/pkg/blob/master/apis/duck/v1/podspec_types.go#L41) [duck typing](https://en.wikipedia.org/wiki/Duck_typing) helps users to configure podSpec fields.
+###### Note: Duck typing in computer programming is an application of the duck test—"If it walks like a duck and it quacks like a duck, then it must be a duck"... -Wikipedia 
+
+
+Right now triggers eventlistener focused to support Kubernetes Deployment and Knative Service,
+But in future there may be possibility to support new CRD (ex: `foo.bar.com`) so to make it standardized implementation
+created the following contract,
+So whoever implements a new CRD in order to support triggers that CRD should satisfy below contract
+
+### Contract
+
+#### Spec
+
+```Spec
+     spec:
+       template:
+         metadata:
+         spec
+```
+
+#### Status
+```staus
+type EventListenerStatus struct {
```

Comment:
Is there a standardized status condition that the EL will report? Or will a user have to figure the right status condition based on whether it is Knative service or regular deployment or something else?

Reply:
The specified `EventListenerStatus` is the one which currently in use so i am preferring to go with same status for any type(like K8s, Knative or something)